apps/web-client/src/lib/stores/inventory/__tests__/display_name.test.ts

```diff
@@ -1,84 +1,141 @@
-import { describe, test, expect } from 'vitest';
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { describe, test, expect, vi } from 'vitest';
 import { writable, get } from 'svelte/store';
+import { testUtils } from '@librocco/shared';
 
 import { NoteTempState } from '$lib/enums/inventory';
 import { NoteState } from '$lib/enums/db';
 
-import type { NoteAppState, NoteStore } from '$lib/types/inventory';
+import type { NoteAppState } from '$lib/types/inventory';
 
 import { createDisplayNameStore } from '../display_name';
 
-import { waitForCondition } from '$lib/__testUtils__/waitForCondition';
+import { newTestDB } from '$lib/__testUtils__/db';
+import { BehaviorSubject } from 'rxjs';
 
-import { defaultNote, defaultWarehouse } from '$lib/__testData__/inventory';
-import { newNote } from '$lib/db/note';
-import { newWarehouse } from '$lib/db/warehouse';
+const { waitFor } = testUtils;
 
 describe('createDisplayNameStore', () => {
-	test('should stream the display name from the content store for given note/warehouse id', async () => {
+	test('should stream the display name from the db for given note/warehouse id', async () => {
+		const db = await newTestDB();
+		const warehouse = await db.warehouse('warehouse-1').create();
+		const note = await warehouse.note().create();
+
 		// Test for note
-		const inNoteStore = writable<NoteStore>({
-			'note-1': defaultNote
+		await note.setName('Note 1', {});
+
+		const ndn$ = createDisplayNameStore(note, null, {});
+		let noteDisplayName: string | undefined;
+		ndn$.subscribe((ndn) => (noteDisplayName = ndn));
```

Comment:
Subscriptions to `ndn$ | wdn$ | pd$ | de$ ...` are never `unsubscribed`?

Reply:
Hmm...interesting point. I honestly don't know if this is a memory leak or if it gets garbage collected when all of the referenced vars are out of scope...